# Why: Preserve the existing optional backend arguments when appending the
# concurrent worker runner to the SPI backend constructor.
diff --git a/engine/src/flutter/shell/common/dart_runtime_backend_factory.cc b/engine/src/flutter/shell/common/dart_runtime_backend_factory.cc
index 2983457cf9d..0943933d6b2 100644
--- a/engine/src/flutter/shell/common/dart_runtime_backend_factory.cc
+++ b/engine/src/flutter/shell/common/dart_runtime_backend_factory.cc
@@ -125,7 +125,8 @@ std::unique_ptr<RuntimeBackend> DartRuntimeBackendFactory::Spawn(
       std::move(backend_image_decoder),
       std::move(backend_image_generator_registry),
       std::move(backend_snapshot_delegate), nullptr, settings.enable_impeller,
-      {}, GetConcurrentWorkerTaskRunner());
+      std::shared_future<impeller::RuntimeStageBackend>{},
+      GetConcurrentWorkerTaskRunner());
   runtime.release();
   return backend;
 }
diff --git a/engine/src/flutter/shell/common/engine.cc b/engine/src/flutter/shell/common/engine.cc
index 80a681f6a79..b6dc497deb9 100644
--- a/engine/src/flutter/shell/common/engine.cc
+++ b/engine/src/flutter/shell/common/engine.cc
@@ -77,6 +77,8 @@ std::unique_ptr<RuntimeBackend> MakeDartRuntimeBackend(
       runtime_delegate, task_runners, runtime->table(),
       std::move(io_manager),
       std::move(image_decoder), std::move(image_generator_registry),
+      fml::TaskRunnerAffineWeakPtr<SnapshotDelegate>{}, nullptr, false,
+      std::shared_future<impeller::RuntimeStageBackend>{},
       concurrent_worker_task_runner);
   runtime.release();
   return backend;
